lib/AST/Decl.cpp

```diff
@@ -4175,10 +4174,12 @@ DeclName AbstractFunctionDecl::getEffectiveFullName() const {
         case AccessorKind::IsAddressor:
         case AccessorKind::IsMutableAddressor:
         case AccessorKind::IsGetter:
-          return subscript ? subscript->getFullName()
-                           : DeclName(ctx, afd->getName(),
-                                      ArrayRef<Identifier>());
-
+          if (afd->getFullName().getKind() == DeclNameKind::Subscript) {
+            return DeclName::createSubscript(ctx, ArrayRef<Identifier>());
```

Comment:
I don't know why I changed this in the first place. I'll change it back to how it was